### PR TITLE
feat: add animations timing control

### PR DESCRIPTION
**Pull request by ogonkov**
Adds global `--yc-animation-multiplier` that controls components animation speed.

### CODE REVIEW

**Review by pseudopirate — APPROVED** [review]

**Review by Marginy605 — APPROVED** [review]

File: styles/styles.scss [hunk 1]
```diff
@@ -22,6 +22,8 @@
     @include yc-brand;
     @include yc-colors-social;
 
+    --yc-animation-multiplier: 1;
```
Comment (ValeraS) [inline styles/styles.scss:25]: Do we need to set it explicitly? We already have the same fallback for it.

File: .storybook/preview.js [hunk 2]
```diff
@@ -62,6 +78,28 @@ export const parameters = {
     },
 };
 
+const animationSpeed = [
+    0.1, 0.2, 0.5, 0.75, 1, 1.25, 1.5, 1.75, 2,
+    // Special value to make speed so fast that it could not be recognized
+    10000,
```
Comment (ValeraS) [inline .storybook/preview.js:84]:
```suggestion
    Infinity,
```

Comment (ogonkov) [reply .storybook/preview.js:84]: We are calculating timing based on this number, we need some real number here

Comment (ValeraS) [reply .storybook/preview.js:84]: `1 / Infinity === 0`

File: styles/styles.scss (hunk 1 above)
Comment (ValeraS) [reply styles/styles.scss:25]: Of cause not, but we set default in fallback and user can change it by setting the var.

File: .storybook/preview.js (hunk 2 above)
Comment (ogonkov) [reply .storybook/preview.js:84]: But we don't need 0, we need some very small value to be sure that animation in JS will trigger

File: styles/styles.scss (hunk 1 above)
Comment (ogonkov) [reply styles/styles.scss:25]: Yes, that is how it should work

File: .storybook/preview.js (hunk 2 above)
Comment (ValeraS) [reply .storybook/preview.js:84]: we need it in storybook?

File: styles/styles.scss (hunk 1 above)
Comment (ValeraS) [reply styles/styles.scss:25]: and why are we setting the var?

File: .storybook/preview.js (hunk 2 above)
Comment (ogonkov) [reply .storybook/preview.js:84]: I think it nice to have all options to see that components works properly with disabled animations

File: styles/styles.scss (hunk 1 above)
Comment (ogonkov) [reply styles/styles.scss:25]: To debug animations in project, for example

File: .storybook/preview.js (hunk 2 above)
Comment (ValeraS) [reply .storybook/preview.js:84]: and animation is disabled when duration is 0. I don't understand, how is this setting in storybook connected to testing animation?

File: styles/styles.scss (hunk 1 above)
Comment (ValeraS) [reply styles/styles.scss:25]: why are we setting it here?

File: .storybook/preview.js (hunk 2 above)
Comment (ogonkov) [reply .storybook/preview.js:84]: As i said before
> we need some very small value to be sure that animation in JS will trigger

File: styles/styles.scss (hunk 1 above)
Comment (ogonkov) [reply styles/styles.scss:25]: To see it dev-tools and adjust it if needed

**Review by korvin89 — APPROVED** [review]

File: .storybook/preview.js (hunk 2 above)
Comment (ogonkov) [reply .storybook/preview.js:84]: How it should look at the component level?

Comment (ValeraS) [reply .storybook/preview.js:84]: Don't use `--yc-animation-multiplier` for critical animation, as was done in this PR for the `toaster` component. Or, for example, add small amount of time to result duration in critical animations.

**Pull request closed without merging**